### PR TITLE
wip: first go at sub categories for documentaion hierarchy

### PR DESCRIPTION
**Pull request by juliacochran**
Screenshot would make more sense if you pretend `BoxGrid` is something like `Getting Started`

screen shot 2016-09-28 at 6 28 21 pm

### CODE REVIEW

File: lib/components/CardPage.js [hunk 1]
```diff
@@ -1,27 +1,30 @@
...
+
   return (
     cards ? <div className="p2">
       {cards.map((card, i) =>
         <div className="mb4" id={`card-${i}`} key={i}>{card.fn()}</div>
        )}
     </div>
     : <div div className="p2">
-      <h1 className="mt0 mb6 display-l dark-gray">Oops!<br/>This page could not be found.</h1>
+      <h1 className="mt0 mb6 display-l dark-gray">Oops!<br />This page could not be found. </h1>
```
Comment (chrislloyd) [inline lib/components/CardPage.js]: Whitespace!

File: init.js [hunk 2]
```diff
@@ -1,32 +1,68 @@
...
-  CURRENT_NAMESPACE = namespace;
-  CARDS[CURRENT_NAMESPACE] = [];
+const MENU = [];
+let currentNamespace = null;
+let currentType = null;
+let currentMenuTitle;
+const NS_TO_MENU_MAP = []; /* Keeps a basic mapping of namespace to menu title for easy retrieval */
+
+function menuItem(type, children = []) {
+  return ({
```
Comment (chrislloyd) [inline init.js]: The `(` isn't necessary here.

File: init.js [hunk 3]
```diff
@@ -1,32 +1,68 @@
...
+    type,
+    children,
+  });
+}
+
+export function registerGroup(groupName, renderPages) {
+  currentMenuTitle = groupName;
+  currentType = 'group';
+  MENU[currentMenuTitle] = menuItem(currentType);
```
Comment (chrislloyd) [inline init.js]: A good practice in this case is to make functions that automatically pass in the `currentType` variable. i.e.

```
groupMenuItem = () => menuItem('group');
```

This means that the implementation is hidden and the fact that only to values are available is enforced.

File: init.js [hunk 4]
```diff
@@ -1,32 +1,68 @@
...
+  renderPages();
+  currentType = null;
+}
+
+export function registerNamespace(ns) {
+  // namespace already exists
+  if (currentType === 'group' && ns in MENU) {
+    throw new Error('You must have unqiue group names and namespaces');
```
Comment (chrislloyd) [inline init.js]: Maybe we should override and warn in this case instead of error?

File: init.js [hunk 5]
```diff
@@ -1,32 +1,68 @@
...
-  CARDS[CURRENT_NAMESPACE].push({
+  MENU[currentMenuTitle].children[currentNamespace].push({
     id,
-    ns: CURRENT_NAMESPACE,
+    ns: currentNamespace,
     fn: () => card,
   });
 }
 
 export function getCards() {
```
Comment (chrislloyd) [inline init.js]: Perhaps we can clarify the naming of this function now? cards / menu etc. are used interchangeably.

File: init.js [hunk 6]
```diff
@@ -1,32 +1,68 @@
...
-  return CARDS;
+  return MENU;
+}
+
+export function getNsToMenuMap() {
```
Comment (chrislloyd) [inline init.js]: I'd prefer to only export one variable — it'll keep it much easier to manage and pass around.

File: lib/components/App.js [hunk 7]
```diff
@@ -7,7 +7,7 @@ export default function App(props) {
   return (
     <div className="md-flex sans-serif" style={{ minHeight: '100vh' }}>
       <div className="col-12 md-col-2 border-box">
-        <Navigation cards={cards} location={location} />
+        <Navigation menu={cards} location={location} />
```
Comment (chrislloyd) [inline lib/components/App.js]: Another example of the cards/menu words being used interchangeably.

File: init.js (hunk 3 above)
Comment (juliacochran) [inline init.js]: Yeah I was thinking that maybe the type should actually be a boolean

File: init.js (hunk 6 above)
Comment (juliacochran) [inline init.js]: should we just return an object with both instead? or you only want the `menu` returned?

File: lib/components/CardPage.js (hunk 1 above)
Comment (juliacochran) [inline lib/components/CardPage.js]: it autofixes the space in `<br/>`. I fixed the other one

File: init.js (hunk 2 above)
Comment (juliacochran) [inline init.js]: fixed

**Pull request merged**